File: backend/src/main/java/com/apocalypse/thefall/entity/character/CharacterCurrentStats.java
package com.apocalypse.thefall.entity.character;


import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Embeddable
@Getter
@Setter
@NoArgsConstructor
public class CharacterCurrentStats {

    @Column(name = "action_points")
    private int actionPoints;

    @Column(name = "hit_points")
    private int hitPoints;

    @Column(name = "level")
    private int level;
}

File: backend/src/main/java/com/apocalypse/thefall/entity/character/CharacterCurrentStats.kt
package com.apocalypse.thefall.entity.character

import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
open class CharacterCurrentStats(

    @Column(name = "action_points")
    open var actionPoints: Int = 0,

    @Column(name = "hit_points")
    open var hitPoints: Int = 0,

    @Column(name = "level")
    open var level: Int = 0
)
